lib/libvarnish/vss.c

```diff
@@ -195,6 +235,44 @@ VSS_resolver(const char *addr, const char *def_port, vss_resolved_f *func,
 	    addr, def_port, func, priv, errp, SOCK_STREAM));
 }
 
+int
+VSS_resolver_range(const char *addr, const char *def_port, vss_resolved_f *func,
+    void *priv, const char **errp)
+{
+	char *p, *hp, *pp;
+	unsigned long lo = 0, hi = 0, i;
+	int error = 0;
+
+	AN(addr);
+	AN(func);
+	AN(errp);
+
+	p = strdup(addr);
+	AN(p);
+	*errp = vss_parse_range(p, &hp, &pp, &lo, &hi);
+	if (*errp != NULL) {
+		free(p);
+		return (-1);
+	}
+
+	if (lo == 0) {
+		/* No port range (0 not allowed in range) */
+		free(p);
+		return (VSS_resolver(addr, def_port, func, priv, errp));
+	}
+
+	for (i = lo; i <= hi && !error; i++) {
+		/* pp points to the first character of the range definition.
+		 * The range definition includes the biggest port number, so the
+		 * buffer must be big enough to fit each number individually.
+		 */
+		sprintf(pp, "%lu", i);
+		error = VSS_resolver(hp, pp, func, priv, errp);
```

Comment:
`hp` here might be `NULL` when `addr` is for example `:5555-5559`, which results in a panic in `VSS_resolver_socktype` (called by `VSS_resolver`) that expects `addr` to never be `NULL`

Reply:
Good catch,  I added a fix and test coverage.

I think it makes sense to support port ranges without addresses